### ISSUE TITLE
Improve API design: proper python library not just CLI

### ISSUE THREAD

**Issue by morenoh149** [Kaggle/kaggle-api#23]
Since many teams collaborate using python notebooks it may be useful to expose a python module interface which allows users to specify their config with variables and allow downloading of competition datasets. Good idea?

**Comment by rysteboe** [comment]
@morenoh149 do you mean being able to use the API within kernels?  If so, that is on my long term to do list, but I don't know when it will be possible.

**Comment by morenoh149** [comment]
no, I'd like to use it within notebooks (colab included) to load datasets into memory. Mainly allowing the library to be used by passing in variables for the secret keys instead of relying on a file with the secret keys (the way it currently works).

**Comment by mirekphd** [comment]
@morenoh149 : maybe you could rename the issue into e.g. "Improve API design: proper python library not just CLI" ?

**Comment by mirekphd** [comment]
The best I could find to merge Kaggle API calls (here: submissions) into my python code was to use os.system(), like this:

```
import os

# submit current predictions file (-f) using its name as annotation (-m)
cmd = 'kaggle competitions submit ' + competition_name + \
    ' -f ' + local_scores_path+cur_preds_file + ' -m ' + cur_preds_file
print(cmd)

# invoke the Kaggle API CLI using a system call
returned_value = os.system(cmd)  

print('Returned value:', returned_value)
assert(returned_value==0)

```
